### PR TITLE
Nicolas - SF Automation - LPD-36750

### PR DESCRIPTION
**Pull request by rubenssoares-br**
Ticket - https://liferay.atlassian.net/browse/LPD-36750
@nicolas-sss

### CODE REVIEW

**Comment by rubenssoares-br** [comment]
ci:test:sf

**Comment by rubenssoares-br** [comment]
ci:test:hotel

**Comment by rubenssoares-br** [comment]
@nicolas-sss

**Comment by rubenssoares-br** [comment]
ci:test:sf

**Comment by rubenssoares-br** [comment]
ci:test:hotel

File: modules/util/source-formatter/src/main/resources/dependencies/replacements.json [hunk 1]
```diff
@@ -3352,6 +3352,22 @@
...
 	},
+	{
+		"classNames": [
+			"Address",
+			"EmailAddress",
+			"OrgLabor",
+			"Phone",
+			"Website"
+		],
+		"from": "regex:(?<variableName>\\w+)\\.((?<prefixMethodName>(get|set))(TypeId))",
```
Comment (nicolas-sss) [inline modules/util/source-formatter/src/main/resources/dependencies/replacements.json:3379]: @rubenssoares-br we could simplify it to this:
`(?<variableName>\\w+)\\.(?<prefixMethodName>get|set)TypeId` ?

It has more parenthesis than necessary. What do you think?

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjsp [hunk 2]
```diff
@@ -25,11 +28,23 @@ PortalUtil.getCurrentAndAncestorSiteGroupIds(themeDisplay.getScopeGroupId());
...
+orgLabor.setListTypeId();
+
+phone.getListTypeId();
+phone.setListTypeId();
+
 ConfigurationProviderUtil.getPortletInstanceConfiguration(clazz, themeDisplay);
 
+website.getListTypeId();
+website.setListTypeId();
+
```
Comment (nicolas-sss) [inline modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjsp:48]: We will have the same problem as before, classNames doesn't work in JSP files, you could create a second ticket as a backlog for when this type of validation is possible. And link with this one. Keep the previous description and add the context that it should also cover JSP files.

Comment (nicolas-sss) [reply modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjsp:48]: @rubenssoares-br

File: modules/util/source-formatter/src/main/resources/dependencies/replacements.json [hunk 3]
```diff
@@ -3352,6 +3352,14 @@
 		"issueKey": "LPD-36648",
 		"to": "getSearchContainer"
 	},
+	{
+		"classNames": [
+			"Contact"
+		],
+		"from": "regex:(?<variableName>\\w+)\\.((?<prefixMethodName>(getPrefix|getSuffix|setPrefix|setSuffix))(Id))",
```
Comment (nicolas-sss) [inline modules/util/source-formatter/src/main/resources/dependencies/replacements.json:3359]: Also can be simplified.

File: modules/util/source-formatter/src/main/resources/dependencies/replacements.json [hunk 4]
```diff
@@ -3360,6 +3360,14 @@
 		"issueKey": "LPD-36750",
 		"to": "${variableName}.${prefixMethodName}ListTypeId"
 	},
+	{
+		"classNames": [
+			"Organization"
+		],
+		"from": "regex:(?<variableName>\\w+)\\.((?<prefixMethodName>(getStatus|setStatus))(Id))",
```
Comment (nicolas-sss) [inline modules/util/source-formatter/src/main/resources/dependencies/replacements.json:3367]: Simplify

**Review by nicolas-sss — CHANGES_REQUESTED** [review]

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjsp (hunk 2 above)
Comment (rubenssoares-br) [reply modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjsp]: Ticket created here: https://liferay.atlassian.net/browse/LPD-37607

Comment (rubenssoares-br) [reply modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjsp]: @nicolas-sss

**Comment by rubenssoares-br** [comment]
ci:test:sf

**Comment by rubenssoares-br** [comment]
ci:test:hotel

**Review by nicolas-sss — APPROVED** [review]

**Comment by drewbrokke** [comment]
Hi all, is this PR still active/relevant? Can we close it?

**Comment by rubenssoares-br** [comment]
cc: @nicolas-sss

**Comment by nicolas-sss** [comment]
Hi, @drewbrokke It is relevant, got blocked due to limitations for JSP cases but a ticket was already created to cover that part, he will rebase, fix conflicts, and resent.

**Comment by rubenssoares-br** [comment]
ci:test:sf

**Comment by rubenssoares-br** [comment]
ci:test:hotel

**Comment by nicolas-sss** [comment]
Hi, @drewbrokke. 
It appears to be a bug in `JSONReplacementsFileCheck` validation, its logic isn't prepared for the regex case and is failing the SF ci test. 

What should we do in this case? Keep this ticket blocked until this problem gets fixed? 
Can you or @kyle-miho create a ticket for it? 

Thanks in advance

**Comment by drewbrokke** [comment]
@nicolas-sss yes, sorry about the delay here. @kyle-miho would you be willing to look into the failure on this one? We can follow up later on whether you feel a new ticket is needed for this case.

**Comment by kyle-miho** [comment]
Fix added here: https://github.com/liferay-devtools/liferay-portal/pull/305

**Pull request closed without merging**